### DISCUSSION TITLE
Website Enhancements Complete: Seeking Additional Feedback

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#3081] [Announcements]
Now that all functional, visual, performance, and security upgrades are complete, I'm curious to hear if there are any other requests for the website. If not, I can return to the regularly scheduled program. 😉

**Comment by oOStroudyOo** [comment]
Remove light theme, it burns my soul :laughing:
Nah I think the website looks great (In dark mode).

**Reply by tteck** [reply]
😆 Agree about the light mode, I'll work on making it less invasive.

**Reply by jesusfj710** [reply]
Please, do not remove the light mode. At least, should be optional. Its quite hard for some people to use a dark mode.

**Reply by tteck** [reply]
Light mode is **not** going to be removed.

**Reply by tteck** [reply]
@jesusfj710 the new/old site now has a light theme.

**Comment by tesla-76** [comment]
Beautiful, I like the clean design. 👍
Perhaps I would add that to the Github discussion page in the links as well.
Still, it's a nice job, congratulations.

**Reply by tteck** [reply]
I did have Discussions linked on the old site. 👍🏻 

**Edit: Added**

**Comment by CrazyWolf13** [comment]
Possibly change the website, so if something is seached result scripts automatically appear on the right, instead of each one needing to be manually clicked. A bit like before.

Otherwise the new website is absolutely amazing!

**Reply by tteck** [reply]
The old search behavior is not possible with the new website, Sorry.

**Comment by jcastro** [comment]
I noticed some spacing issues depending on the size of the browser window. Attaching some screenshots!

<img width="1392" alt="SCR-20240528-lifn" src="https://github.com/tteck/Proxmox/assets/190036/2aab14f3-8c43-4d6a-ac04-36459c5bb8b5">
<img width="965" alt="SCR-20240528-lilu" src="https://github.com/tteck/Proxmox/assets/190036/cc4cfdc8-2b96-46c8-8e5b-ac0f0ae5714f">

**Reply by tteck** [reply]
Thanks, noted. What are your opinions on the "Light" theme?

**Reply by tesla-76** [reply]
I put it in Light, I would test the top band where the links are in gray like the LXC container boxes.  It would thus lower the glare effect of the whole white page.

In full screen, as reported by the other user, the left and right spacing is missing (a minimum of pixels off the edge of the window would increase the readability especially of the name column on the left).

**Reply by tteck** [reply]
Must be browser specific,  I use Chrome and don't experience those issues.

**Reply by tesla-76** [reply]
I also use Chrome but the space on the sides at full screen is missing. Now I'm going to try Edge.

P.S. Edge is also lacking space.

**Reply by tteck** [reply]
If I slowly narrow the browser, I see the margins start to decrease until no margin exists, then it starts to wrap the text and the side margins reappear. Is this the behavior you're describing? The moment before word wrap.

**Reply by tesla-76** [reply]
If I put the browser in the window and slowly widen it to 80% of the monitor width the spacing remains. Then it disappears and the text sticks to the edge. It's witchcraft!😆

**Reply by tteck** [reply]
Refresh the browser and check again, I made some changes to always maintain _some_ side margins.

**Reply by tesla-76** [reply]
> Refresh the browser and check again, I made some changes to always maintain _some_ side margins.

👍Tested on Chrome, now is ok. The anti-witch ritual was successful!

**Reply by tteck** [reply]
🤣 Good!

**Reply by jcastro** [reply]
> Thanks, noted. What are your opinions on the "Light" theme?

I think it's nice to have the option; usually, I set it to System, as it changes depending on the time of day.

One thing I would personally do is to remove or decrease the opacity on these backgrounds so there's better contrast with the text. Basically, just having an outline without a background, for example.

<img width="921" alt="image" src="https://github.com/tteck/Proxmox/assets/190036/136f3844-7670-40cf-9a8d-cd7369f93edd">
<img width="926" alt="image" src="https://github.com/tteck/Proxmox/assets/190036/7a5ae4cb-0489-45bd-ba52-276f5fd91712">

**Comment by skidoodle** [comment]
I'd still like the idea of highlighting the selected script's name on the left along with the onhover thing. (https://github.com/skidoodle/proxmox-helper-scripts/commit/c40f5b9f853883fb190c732b9c0bfe579872c34b)

**Reply by tteck** [reply]
I tried it, but found it unnecessary since the selected script was shown on the right. It's just a matter of preference.

**Comment by tteck** [comment]
Add "View Source" to "Newest Scripts"

![Screenshot 2024-05-28 5 54 07 PM](https://github.com/tteck/Proxmox/assets/21342670/c7022150-4685-4794-8407-36fcde644cdf)

**Reply by oOStroudyOo** [reply]
This is so neat!!
I no longer need to go searching for it in Github. I tend to study the source to help me learn some stuff.

Awesome addition, thank you.

**Reply by AdamYellen** [reply]
Thank you so much for this feature. I always manually navigated over to Github to look at the source. Not necessarily because I don't trust the script, but because I like seeing how other people do things.

**Comment by mastervash** [comment]
What about having a small counter for number scripts?
To the right... or maybe just above 'Categories'

"_Proudly serving_ X _scripts_."

**Reply by tteck** [reply]
Added a count of scripts per category. I figured this would be more useful than just a total of all.

![Screenshot 2024-06-01 8 56 25 PM](https://github.com/tteck/Proxmox/assets/21342670/a9ec098a-bf77-421f-8ea1-3ec9560046e5)

**Reply by tteck** [reply]
Heck, why not display both.

**Comment by shakebox** [comment]
I really like the new page from a user perspective. but there is one small (or bigger?) downside: it's very heavy on the CPU, at least for me. No matter if I use current Firefox or Chrome on my Macbook Pro M1Pro, it's one of very few pages that after a while brings my machine to feel hot and the fan to spin and make noise. Even the first startpage, but even more the actual scripts page. most likely there because of the 'logo snow' overlay!?
Of course I don't need to have the page open for long times. but checking the ressource usage would be really welcome

**Reply by oOStroudyOo** [reply]
Just tried monitoring this myself.
The CPU momentarily jumps. I was at about 3% and it jumped to 10% for about 2 seconds and then returned to 3%.
RAM almost didn't move.

Processor: Intel Core i7-10850h

**Comment by cobraf0** [comment]
I seem to have an issue loading the website content from my home network
When i try from my phone connected to cell it works perfectly.
Looks like the calls to https://pocketbase-proxmox-helper-scripts.fly.dev/api/collections/categories/records?page=1&perPage=500&skipTotal=1&expand=items&sort=order fail with timeout
Is it possible there is IP blocking going on on this website?

**Reply by tteck** [reply]
Do you have any firewall rules set? DNS issue?

**Reply by cobraf0** [reply]
i am using cloudflare DNS on my router which runs Opnsense
You can see the resolves from DNS below, the helper-scripts.com website loads but only has the iframe with the title and the search on the left, the whole middle is empty.

When i look in Dev console i see that the requests to pull the tiles fail with a timeout
I checked the firewall i dont see any blocks to these requests.
I also tried the phone on wifi which doesnt work, turning off the wifi fixes the issue.

This is why i thought there might be ipblocking on the pocketbase-proxmox-helper-scripts.fly.dev domain

Non-authoritative answer:
Name:    helper-scripts.com
Address:  76.76.21.21

Non-authoritative answer:
Name:    pocketbase-proxmox-helper-scripts.fly.dev
Addresses:  2a09:8280:1::33:f29c:0
          66.241.124.26

**Reply by cobraf0** [reply]
ok so i found the issue, when i disable IPV6 on my pc it works
It seems that Name: pocketbase-proxmox-helper-scripts.fly.dev resolves to both IPv6 and IPv4 and there is some kind of an issue with IPv6

**Reply by cobraf0** [reply]
it might be a good idea to check why the dns has ipv6 and if its intended make sure the configuration accepts ipv6 connections on the site

**Reply by tteck** [reply]
Would you check again to see if this issue still exists?

**Reply by cobraf0** [reply]
retested, works now

**Comment by oOStroudyOo** [comment]
I find the View Source feature very useful.

Would it be possible to add this as a button on all apps rather than the latest 3?

**Reply by tteck** [reply]
Yes, it's possible.

**Reply by tteck** [reply]
"View Source" is now located within "Selected Script"

![Screenshot 2024-06-02 2 58 22 AM](https://github.com/tteck/Proxmox/assets/21342670/772faa90-b455-45ec-8071-bd64f7441d38)

**Reply by tteck** [reply]
That gives the user an easy way to inspect the code before running the script.

**Comment by pythongod** [comment]
The floating logos in the background are so irritating and distracting, especially in the main working area (categories and selected scripts). If they only touch the margin space / free space, it would be okay/better but that will be some hard css I imagine.

**Reply by tteck** [reply]
Thank you for your feedback. At this time, there are no plans to change the floating logos in the background.

**Comment by joon-im** [comment]
I think it would be cool if there was some way to scroll through the newest and most-viewed scripts. Or maybe a link to a new page to see more of these scripts? I like seeing what others are installing.

**Reply by tteck** [reply]
Thank you for your feedback! You can stay updated on the latest changes (beyond the three newest scripts) in the repository by following the [changelog](https://github.com/tteck/Proxmox/blob/main/CHANGELOG.md). While the "Most Viewed" section gives you an idea of popular scripts, it doesn't necessarily reflect the most installed ones.

**Reply by joon-im** [reply]
I didn't even know that existed! Thank you.

**Comment by Masked-Kunsiquat** [comment]
Loving the updated site! 
only recommendation (and it may be because I'm on mobile) - could you put a link to the GitHub on the site? 


Google now shows helper-scripts as the top hit (YAY!); but I currently have to search for the GitHub... might make it easier for new users to find the discussions and such (:

thanks for all your amazing work!!

**Reply by tteck** [reply]
![Screenshot 2024-06-04 4 46 59 AM](https://github.com/tteck/Proxmox/assets/21342670/e4a8c488-83ec-43f8-bcd9-6a97eabab5be)

**Comment by tteck** [comment]
Important notices will now be displayed on the landing page.

![Screenshot 2024-06-06 4 57 22 PM](https://github.com/tteck/Proxmox/assets/21342670/2c01040d-85d2-4c68-9a3e-82bbff49bc21)

**Reply by ndom91** [reply]
Is the website opensource? I can't find the repository. I'd love to make some PR's, for example improving the marquee banner :+1:

**Reply by tteck** [reply]
No, the source is private.

**Reply by tteck** [reply]
99% of the time that banner will not be needed, hopefully. 🤞🏻

**Reply by tteck** [reply]
@ndom91 https://github.com/tteck/Proxmox/discussions/2889 https://github.com/tteck/Proxmox/discussions/3042

**Comment by tteck** [comment]
Since many users bookmark https://helper-scripts.com/scripts, they may not be aware of any important messages. Therefore, I've added a notification about important messages. Once users acknowledge the message, it won't show up again unless the browser cache is cleared or thirty days have passed.

![Screenshot 2024-06-07 9 39 25 PM](https://github.com/tteck/Proxmox/assets/21342670/405e41a5-5af4-4c5b-81e3-03affbf195fa)

**Comment by theminkkilledcuriosity** [comment]
New layout looks and interacts great! Just verifying there are a few other scripts on the github that are not on the website?

**Reply by tteck** [reply]
yep

**Reply by tteck** [reply]
15 if I counted correctly.

**Reply by theminkkilledcuriosity** [reply]
awesome was just checking! No complaints. The new layout looks good, plus no need to bust your ass over a website compared to all the effort you have put into the community! haha I appreciate all you have done! I will keep an eye out and will alert you if I encounter any bugs!

**Comment by tteck** [comment]
I reopened the gh-pages site (https://tteck.github.io/Proxmox/) to alleviate some traffic from helper-scripts.com. Given the amount of traffic the website receives, it will cost between $80 and $100 per month. Although it's nice, I don't think the new site on Vercel is worth that!

**Reply by oOStroudyOo** [reply]
Try Cloudflare for hosting.
As long as you comply with their Terms of Service I don't believe they have bandwidth limits.

I've been using the free plan for years and pointed the domain to my Github Repo for webpages.

Other than the price of a domain, I've never spent a single penny on it.

**Reply by tteck** [reply]
Yeah, it used to be $20 a month and no bandwidth limits, now it's like Vercel.

**Reply by tteck** [reply]
gh-pages is free, period.  I'll work on the gh-pages site to make it better. You can link apps now, only open one category at a time, plus, we have the old style search back.

**Reply by DomJeal** [reply]
New site is nice and flashy but as you said, not worth $100 pcm!

Bookmark changed back, the gh-pages site is brilliant as is, the only thing I wish is that the search bar was in focus on load of the page.

**Reply by svengreb** [reply]
I'm not sure how it was deployed before, but you can still simply use GitHub Pages by [exporting the Next.js sites statically](https://nextjs.org/docs/app/building-your-application/deploying/static-exports) which allows to host it on any dump HTTP server. Yes, you must adjust your site's code in some aspects to not use any server-side logic, but for a project like this static rendering is totally fine and updates can be automated with GitHub _Actions_ workflows.
Therefore, using a commercial service like Vercel is not required in any way for static content.

**Reply by tteck** [reply]
I'd love to learn more. If you use Discord, my username is tteckster. I would love to discuss this further.

**Reply by jedikaal** [reply]
Thank you for all your hard work. Because of this I am able to spent enough time to setup services while not taking to long so I do not loose my attention span. 

The upgrade website was cool but the current (old) one is really functional. Because it is free I would say. Keep the money you earn and invest it in updating the scripts and making new one. 

Keep the loving work going! ❤️

**Reply by svengreb** [reply]
@tteck Sure, it's just a matter of time because busy adult life things keeps me busy.

**Reply by mh166** [reply]
@svengreb That annoying adult life all the time... I mean, we were never asked if we really wanted it anyways, right? 😉 But seriously, I bet everyone can understand. So just take your time as needed. 😊

However, I also wanted to say how much I would appreciate your effort to make the new site available again. Not to push, mind you, but just cheering from the side line! 🥳

**Reply by tteck** [reply]
I'm beginning to prefer the current site now. Fast, simple and free. Also the search feature is much better.

**Comment by tesla-76** [comment]
Too bad because the new site had succeeded really nice, but the maintenance cost is really expensive. But more than the "Buy me a coffee" button it would take the "Bye me a pizza menu" button to keep it active! 😄

For the good old site, if it is possible, I would suggest making the names of the programs in the changelog links to the tab where to download the relevant script.

**Reply by wisie** [reply]
Sad to see the new website go but totally makes sense to why.

One thing I liked about the new website was the ease to see what's changed. Upvote for this idea of linking in the changelog.

**Comment by SiskoUrso** [comment]
That is a bummer, was really liking the new site and was confused to see it go back to the old look. Came searching for some info here, agreed that is allot to host it compared to just using the free pages with github.
